File: Problemas_python/problema3.py
# Problema 3 :  Realizar una divisición 

try:
    numero1 = int(input("Ingrese el dividendo: "))
    numero2 = int(input("Ingrese el divisor: "))
    resultado = numero1/ numero2
    print(f"El resultado de la división es: {resultado}")
    
except ZeroDivisionError: #Excepción para división por cero
    print("Error: El divisor no puede ser cero.")

File: problemas_haskell/problema3.hs
-- Problema 3 :  Realizar una divisición 

division :: Int -> Int -> Either String Int
division _ 0 = Left "Error: El divisor no puede ser cero."
division numero1 numero2 = Right (numero1 `div` numero2)

main :: IO ()
main = do
    putStrLn "Ingrese el dividendo:"
    aStr <- getLine
    putStrLn "Ingrese el divisor:"
    bStr <- getLine
    let num1 = read aStr :: Int
        num2 = read bStr :: Int
    case division num1 num2 of
        Left err -> putStrLn err
        Right res -> putStrLn $ "El resultado de la división es: " ++ show res --Solo el numero entero